packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,226 @@
+contract LivenessModule2 is ILivenessModule2 {
+    /// @notice Reserved address used as the previous owner to the first owner in a Safe
+    address public constant SENTINEL_OWNER = address(0x1);
+
+    /// @notice Mapping from Safe address to its configuration
+    mapping(address => ModuleConfig) public safeConfigs;
+
+    /// @notice Mapping from Safe address to active challenge start time (0 if no challenge)
+    mapping(address => uint256) public challengeStartTime;
+
+    /// @notice Semantic version.
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
+
+    /// @notice Configures the module for a Safe that has already enabled it
+    /// @dev MUST only be callable by a Safe that has enabled this module
+    /// @param _config The configuration parameters for the module
+    function configure(ModuleConfig memory _config) external {
+        if (_config.livenessResponsePeriod == 0 || _config.fallbackOwner == address(0)) {
+            revert LivenessModule2_InvalidParameters();
+        }
+
+        // Check that this module is enabled on the calling Safe
+        Safe safe = Safe(payable(msg.sender));
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        // Store the configuration for this safe
+        safeConfigs[msg.sender] = _config;
+
+        // Clear any existing challenge when configuring/re-configuring
+        // If a challenge exists, it MUST be canceled, including emitting the appropriate events
+        if (challengeStartTime[msg.sender] != 0) {
+            delete challengeStartTime[msg.sender];
+            emit ChallengeCancelled(msg.sender);
+        }
+
+        emit ModuleEnabled(msg.sender, _config.livenessResponsePeriod, _config.fallbackOwner);
+    }
+
+    /// @notice Clears the module configuration for a Safe
+    /// @dev MUST only be executable by a Safe that has DISABLED this module first
+    /// @dev MUST erase the existing liveness_response_period and fallback_owner data related to the calling safe
+    /// @dev Note: Clearing the configuration also cancels any ongoing challenges
+    function clear() external {
+        // Check if the calling safe has configuration set
+        ModuleConfig storage config = safeConfigs[msg.sender];
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+
+        // Check that this module is NOT enabled on the calling Safe
+        // This prevents clearing configuration while module is still enabled
+        Safe safe = Safe(payable(msg.sender));
+        if (safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleStillEnabled();
+        }
+
+        // Erase the configuration data for this safe
+        delete safeConfigs[msg.sender];
+        // Also clear any active challenge
+        delete challengeStartTime[msg.sender];
```

Comment:
You need to emit a `ChallengeCancelled` event if there was a challenge and you deleted it here. Basically the same `if` block as in `configure`. It could be an internal function.

Reply:
An internal `_cancelChallenge()` function that bundles the deletion with the event would be nice for helping to encourage this pattern.

Reply:
Fixed in 1c27cb67bd3974ac417b38c63975ac506347231f